### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by jasmineliaw** [pola-rs/polars#5473]
### Problem description

Propose to implement/modify functions below which currently are not supported in Polars or has limitations:
1. `pd.to_numeric`
- Use case: convert all values to numeric with different methods provided to deal with invalid parsing
- https://pandas.pydata.org/docs/reference/api/pandas.to_numeric.html
2. `resampler.nearest`
- https://pandas.pydata.org/docs/reference/api/pandas.core.resample.Resampler.nearest.html#pandas.core.resample.Resampler.nearest
3. Allow to set tz=None in `tz_localize`
- Pandas: https://pandas.pydata.org/docs/reference/api/pandas.Series.tz_localize.html
- Polars: https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.dt.tz_localize.html
4. `pivot` function is not available in `LazyFrame`
- https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.DataFrame.pivot.html
5. `dropna(how="all", axis=1)` to drop columns with all values are null
- Example in https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.DataFrame.drop_nulls.html works for `polars.DataFrame` but not `LazyFrame`
6. Select columns by conditions
- Use case: select columns from `LazyFrame.var` where all values > 0
- https://pola-rs.github.io/polars/py-polars/html/reference/lazyframe/api/polars.LazyFrame.var.html

**Comment by ritchie46** [comment]
Pivot will not be exposed in lazy because we cannot determine the schema without collecting the data first. So you must do the collect explicit.

Regarding your feature requests, could you split them up? Pushing a list of wishes here is not how we order feature requests. We'd like to have discussion per feature and some of the mentions here are already supported, but in another syntax.

**Comment by ritchie46** [comment]
I will close this and await the detailed requests per topic.

**Comment by jasmineliaw** [comment]
Thanks @ritchie46

I have created separate issues for points 1,2,5,6:
1. pd.to_numeric: #5494 
2. resampler.nearest: #5495
3. Allow to set tz=None in tz_localize: Found the syntax to deal with this - use `dt.with_time_zone`
4. pivot function is not available in LazyFrame: Answered
5. dropna(how="all", axis=1) to drop columns with all values are null: #5496 
6. Select columns by conditions: #5496 

Looking forward to your advice. Thank you!
